src/debug/HyprCtl.hpp

```diff
@@ -29,4 +29,8 @@ class CHyprCtl {
     wl_event_source*                 m_eventSource = nullptr;
 };
 
+std::string                      getWindowData(PHLWINDOW w, eHyprCtlOutputFormat format);
```

Comment:
hm, I don't like leaving globals like this, without any namespace or anything.

Does anyone need those outside of the one cpp file anyways? Why should they be exposed?

Reply:
In my case, I'm developing a plugin for Hyprland that acts as a proxy for events. This plugin allows you to run scripts in Bash, Python, etc., when event hooks are called. This makes some dotfile functionality possible without having to write a plugin each time you got an idea. To enhance functionality, I want to pass the JSON representation of windows, monitors, etc., as a parameter to these scripts.

Since there's already maintained code for generating these JSON representations, it would be redundant and inefficient for me to write it from scratch, given it means copying the existing code from hyprctl.cpp and maintaining it myself. I'm looking for better options to make this accessible from outside the plugin. Any suggestions for a more efficient solution would be greatly appreciated.

Reply:
One possible option would be moving the functions to the class and making them static so they can get called through the `g_pHyprCtl` pointer.

Reply:
"Why should they be exposed?"

Because you write a plugin that hooks something in that file, like say you want to slightly reimplement it to add some extra data or whatever. But the function uses a bunch of 'private' functions in HyprCtl.cpp so you either have to cut+paste them or use pointers to them or something similarly ugly.

If they were just in a namespace at least you could get to them somehow.

Reply:
fair enough.

Just move it somewhere that isnt global then, please.